src/errors.ts

```diff
@@ -0,0 +1,87 @@
+import { AxiosError } from 'axios'
+
+export class ConnectionError extends Error {
```

Comment:
nit: maybe move it with the other errors as the enum makes a big separation

Reply:
Put the enum at the bottom